### DISCUSSION TITLE
Trouble installing ketcher-react

### DISCUSSION THREAD

**Discussion by mwbouwkamp-mendix** [epam/ketcher#1183] [Q&A] [answered]
When I try to use the ketcher-react npm package, I get the following error upon installation:

    npm ERR! code EUNSUPPORTEDPROTOCOL
    npm ERR! Unsupported URL Type "workspace:": workspace:*

This the package I am referring to: https://www.npmjs.com/package/ketcher-react

Any suggestions?

**Comment by AndreiMazol** [comment, marked as answer]
Hi @mwbouwkamp-mendix ,

 the issue has been solved yesterday. please use the latest  version of kecther-react and its dependencies.

Best regards,
Ketcher team

**Reply by mwbouwkamp-mendix** [reply]
Thank you for your answer. I am now indeed able to install the package!

Could you verify that the manual is still up to date? Now I get an error "cannot find name process" when I try the code below and I am prompted to add an additional import for the Editor.


```
import { RemoteStructServiceProvider } from 'ketcher-core'

const structServiceProvider = new RemoteStructServiceProvider(
  process.env.REACT_APP_API_PATH!
)

const MyComponent = () => {
  return (
    <Editor
      staticResourcesUrl={process.env.PUBLIC_URL}
      structServiceProvider={structServiceProvider}
    />
  )
}
```

**Reply by AndreiMazol** [reply]
This code is copied from /example folder. You can use any  string value to  initialize RemoteStructServiceProvider or just use standalone version. For details please look at https://github.com/epam/ketcher/tree/master/example

**Reply by mwbouwkamp-mendix** [reply]
Thank you very much!

**Comment by mwbouwkamp-mendix** [comment]
Hello @AndreiMazol : It seems I am still running into issues. When creating a build of my project, I get the following error:

`Error: 'url' is imported by url?commonjs-external, but could not be resolved – treating it as an external dependency`

Any suggestions?

**Reply by AndreiMazol** [reply]
Hi @mwbouwkamp-mendix ,

As far as I understand you  use rollup  for building your App, so  I guess it is necessary to update your rollup.config.js. As an option you  can use [CRA](https://create-react-app.dev/) instead of configuring your app manually.

Regards,
Andrei

**Comment by luna215** [comment]
When I run npm install --save ketcher-react, I get this following error:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: molstar@3.13.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   dev react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.2" from ketcher-react@2.5.2
npm ERR! node_modules/ketcher-react
npm ERR!   ketcher-react@"^2.5.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/paulluna/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/paulluna/.npm/_logs/2022-09-13T00_22_25_935Z-debug-0.log
```

Has anyone else had trouble installing this?
